datalab/datalab_session/utils/filecache.py

```diff
@@ -0,0 +1,227 @@
+import os
+import time
+import logging
+from pathlib import Path
+
+from django.core.cache import cache
+from django.conf import settings
+
+from datalab.datalab_session.utils.s3_utils import download_fits
+
+log = logging.getLogger()
+log.setLevel(logging.INFO)
+
+
+class FileCache():
+    ''' Class for managing file access. It uses django_redis to support a LRU cache of files in redis.
+        It is meant to be a drop-in non-contextmanager replacement for the get_fits method we previously
+        used, as well as having an add_file_to_cache method which adds a created dataproduct that is on
+        the filesystem to the file cache.
+    '''
+    LOCK_TIMEOUT = 5  # lock timeout in seconds
```

Comment:
I just picked this randomly, but its basically the time for which the lock will be active. We want it long enough to do the operations within a lock, which are usually just retrieving or setting a few keys in redis, but not so long that if the container dies the lock will block a newly spawned container from doing anything.